### ISSUE TITLE
LWFA simulation with mesh refinement gives an instability when warpx.do_subcycling = 1

### ISSUE THREAD

**Issue by prkkumar** [ECP-WarpX/WarpX#3345]
In an LWFA simulation with mesh refinement, I noticed an instability when I use `warpx.do_subcycling = 1`. The issue can be reproduced with the following input script:
[input.txt](https://github.com/ECP-WarpX/WarpX/files/9552530/input.txt)

Instability seems to appear inside the fine patch. Note the high field values inside the fine patch in the plot below (Top left: Wakefield, Top right: Laser's field, Bottom left: total charge density, Bottom right: Ion density)
<img width="700" alt="subcycling" src="https://user-images.githubusercontent.com/89051199/187108836-e61e3f0a-64bb-4be3-99a1-f46195f44eb3.png">.

When I turn off subcycling, I DO NOT see the instability. Plots below are obtained with `warpx.do_subcycling = 0`
<img width="853" alt="NoSubcycling" src="https://user-images.githubusercontent.com/89051199/187109133-4071e430-4617-429c-820a-9a0a76ff59c2.png">

CKC solver is used in these simulations.

Note that, for simulations with coarser resolution in the transverse direction (`nx = 128` and `nx = 256`), I do not see any instability. It only begins to appears for `nx = 512`.

**Comment by prkkumar** [comment]
Following the suggestion in the Monday meeting on 08/29/2022, I modified the `amr.max_grid_size` and `amr.blocking_factor` to make sure fine patch grids are "properly aligned". With 

`amr.max_grid_size = 2048 2048`
`amr.blocking_factor = 128 128`

I DO NOT see the instability and the results make sense. Here is the `Ez` field with these changes.
![Ez_subcycling](https://user-images.githubusercontent.com/89051199/189757314-8b14ee0d-624f-43dd-8258-f2b4f468d9b4.png)

**Comment by prkkumar** [comment]
Now, I see an instability with `nx = 1024` (two times the previous case reported above). This time, choosing a blocking factor to align the grids does not solve the problem. See the Ez field below. 
![Screen Shot 2022-10-02 at 5 03 15 PM](https://user-images.githubusercontent.com/89051199/193482984-eecfc129-d39f-44e5-b737-106d9ebc1eea.png)

The input script that was used is attached here:
[input_instability.txt](https://github.com/ECP-WarpX/WarpX/files/9693892/input_instability.txt)

Job submission script is attached here:
[submission_script.txt](https://github.com/ECP-WarpX/WarpX/files/9693893/submission_script.txt)

**Comment by RemiLehe** [comment]
@prkkumar Recently shared with me a case which is unstable even with `nx = 512`: 
[inputs.txt](https://github.com/ECP-WarpX/WarpX/files/9721023/inputs.txt)
![Screen Shot 2022-10-05 at 8 55 39 PM](https://user-images.githubusercontent.com/6685781/194210596-f2b17cdf-4c53-4cda-bb28-d85eb9d9ef2a.png)

**Comment by RemiLehe** [comment]
Unfortunately, it looks like this the bug does not show up in 2D

**Comment by RemiLehe** [comment]
Unfortunately, it looks like this the bug does not show up in 2D:
[inputs.txt](https://github.com/ECP-WarpX/WarpX/files/9721080/inputs.txt)


![Screen Shot 2022-10-05 at 9 07 42 PM](https://user-images.githubusercontent.com/6685781/194211787-dc9b2d5d-9df9-4b1f-ae33-dd85a71025f4.png)

**Comment by RemiLehe** [comment]
Interestingly, it looks like the bug is suppressed (in 3D) when changing 
```
amr.blocking_factor_x = 128 64 
amr.blocking_factor_y = 128 64
```
to
```
amr.blocking_factor_x = 128 128
amr.blocking_factor_y = 128 128
```
![Screen Shot 2022-10-05 at 9 00 16 PM](https://user-images.githubusercontent.com/6685781/194211064-7d9125be-b4a3-4516-bfd1-80c7d2112e59.png)

**Comment by RemiLehe** [comment]
I also tried removing PMLs with:
```
do_pml_Lo[lev] = amrex::IntVect::TheZeroVector();
do_pml_Hi[lev] = amrex::IntVect::TheZeroVector();
```
in `Source/Initialization/WarpXInitData.cpp`, but it still shows the instability.

**Comment by RemiLehe** [comment]
Interestingly, the instability is also removed when using decomposition which is such that the MR patch is not transversally decomposed:
![Screenshot from 2022-10-06 14-43-43](https://user-images.githubusercontent.com/6685781/194427723-3861e4af-7809-4f47-a839-e3fc0ac5e9fe.png)
This was achieved with:
```
amr.max_grid_size_x = 128 128
amr.max_grid_size_y = 128 128
amr.max_grid_size_z = 128 128
amr.blocking_factor_x = 128 64
amr.blocking_factor_y = 128 64
amr.blocking_factor_z = 128 128
amr.refine_grid_layout_x = 0
amr.refine_grid_layout_y = 0
```

**Comment by RemiLehe** [comment]
This issue is fixed by #3455.
